Let's a user share their WebID via a QR code

brianscherner opened the pull request:
## This PR:
Resolves #691 
 
**1. Added a new button labeled "Show QR Code" on the user's Profile page, which opens a modal containing a scannable QR code of the user's webID using a library called _react-qr-code_. Their WebID is also displayed underneath the code for easy verification.**    

## Screenshots (if applicable):
![qr-code-btn](https://github.com/user-attachments/assets/96ccda58-69fb-4a03-b9c5-5f395de4e51e)
![qr-code-modal](https://github.com/user-attachments/assets/62275872-7f04-4509-8fd3-3a7c2f804803)

## Additional Context (optional):
This has currently only been tested on a _localhost_ WebID but should work with other pods.

## Future Steps/PRs Needed to Finish This Work (optional):
- Could add a _Download_ button so the QR code can be saved as a PNG for easy sharing (although they could just screenshot it as well - might not be necessary).
- Consider adding QR code support for other areas (ex: contact sharing).

brianscherner committed:
WIP commit for creating a QR code to share webId with others.

brianscherner committed:
Display QR code in a modal.

brianscherner committed:
Make 'close' button width responsive.

brianscherner committed:
Change location of QR code feature and style as button.

brianscherner committed:
Update build artifacts.

brianscherner committed:
Update upload-pages-artifact to v2.

leekahung commented:
I see that this PR is based on #706.

I would merge the other PR first into development then rebase this branch to line up with Development.

Should be good to review after that.

leekahung commented on src/components/Profile/CreateQRCode.jsx:
```diff
@@ -0,0 +1,93 @@
+ *
+ * @param root0
+ * @param root0.webId
+ */
```
Would update this to be a proper JSDoc comment, check other components for reference. The Modal components have very nice examples to follow.

leekahung commented:
The feature looks good, had just checked it on my local machine.

leekahung commented:
I've found a solution to the failing check in the CI/CD process, see #708. Think we can revert some of the changes you made to `ci.yml` here to avoid merge conflicts.

brianscherner committed:
Revert commits made to ci.yml.

This reverts commit 664105078afe4e08826eafb6afc077daf58e37ad.

brianscherner committed:
Update component documentation.

brianscherner committed:
Merge branch 'Development' into 691-qr-code-enhancement

brianscherner commented:
I updated the component's documentation based on the examples from the other modal components and reverted the changes I had made to ci.yml.

I'm still getting a failed build check, but I think that's because this branch doesn't have the updated NPM package patches from #708 yet

leekahung commented:
> I'm still getting a failed build check, but I think that's because this branch doesn't have the updated NPM package patches from #708 yet

Yeah, let me merge the other PR. Think that when we update this branch with the updates from #708, we should pass the checks

leekahung committed:
Merge branch 'Development' into 691-qr-code-enhancement

leekahung reviewed (approved):
Sweet! I think we're in the clear.
